crates/turborepo-ui/src/tui/preferences.rs

```diff
@@ -0,0 +1,172 @@
+use serde::{Deserialize, Serialize};
+use turbopath::AbsoluteSystemPathBuf;
+
+const TUI_PREFERENCES_PATH_COMPONENTS: &[&str] = &[".turbo", "preferences", "tui.json"];
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+    #[error(transparent)]
+    Serde(#[from] serde_json::Error),
+}
+
+pub struct PreferenceLoader {
+    file_path: AbsoluteSystemPathBuf,
+    config: Preferences,
+}
+
+impl PreferenceLoader {
+    pub fn new(repo_root: &AbsoluteSystemPathBuf) -> Result<Self, Error> {
+        let file_path = repo_root.join_components(TUI_PREFERENCES_PATH_COMPONENTS);
+        let contents = file_path.read_existing_to_string()?;
+        let config = contents
+            .map(|string| serde_json::from_str(&string))
+            .transpose()?
+            .unwrap_or_default();
+
+        Ok(Self { file_path, config })
+    }
+
+    pub fn is_task_list_visible(&self) -> bool {
+        self.config.is_task_list_visible.unwrap_or(true)
+    }
+
+    pub fn set_is_task_list_visible(&mut self, value: Option<bool>) {
+        self.config.is_task_list_visible = value;
+    }
+
+    pub fn active_task(&self) -> Option<&str> {
+        let active_task = self.config.active_task.as_deref()?;
+        Some(active_task)
+    }
+
+    pub fn set_active_task(&mut self, value: Option<String>) -> Result<(), Error> {
+        self.config.active_task = value;
+        Ok(())
+    }
```

Comment:
No need to lie to the caller that this could fail. We can propagate this change up to callers that now no longer need to handle the error case.
```suggestion
    pub fn set_active_task(&mut self, value: Option<String>) {
        self.config.active_task = value;
    }
```